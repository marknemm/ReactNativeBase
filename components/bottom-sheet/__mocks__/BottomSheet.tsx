import type { BottomSheetProps } from '@components/bottom-sheet/BottomSheet.interfaces';

const MockBottomSheet = jest.fn().mockImplementation((props: BottomSheetProps) => {
  const ActualComponent = jest.requireActual('@components/bottom-sheet/BottomSheet').default;
  return <ActualComponent {...props} />;
});

export default MockBottomSheet;
