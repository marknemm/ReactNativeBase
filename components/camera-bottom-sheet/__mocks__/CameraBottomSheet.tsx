import type { CameraBottomSheetProps } from '@components/camera-bottom-sheet/CameraBottomSheet.interfaces';

const MockCameraBottomSheet = jest.fn().mockImplementation((props: CameraBottomSheetProps) => {
  const ActualComponent = jest.requireActual('@components/camera-bottom-sheet/CameraBottomSheet').default;
  return <ActualComponent {...props} />;
});

export default MockCameraBottomSheet;
