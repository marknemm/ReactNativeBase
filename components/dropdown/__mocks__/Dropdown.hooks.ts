/**
 * Mocks the useDropdownItems hook.
 */
export const useDropdownItems = jest.fn().mockImplementation((data) => data);
