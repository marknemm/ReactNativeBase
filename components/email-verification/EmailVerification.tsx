import ErrorText from '@components/error-text/ErrorText';
import { useSubmitState } from '@hooks/form-hooks';
import { Button, Icon, Text } from '@rneui/themed';
import { View } from 'react-native';
import type { EmailVerificationProps } from './EmailVerification.interfaces';
import { useStyles } from './EmailVerification.styles';

/**
 * A component for displaying email verification status and resending verification emails.
 *
 * @param props The component {@link EmailVerificationProps}.
 * @returns The {@link EmailVerification} component.
 */
const EmailVerification: React.FC<EmailVerificationProps> = ({
  containerStyle,
  isVisibleWhenVerified = false,
  user,
}) => {
  const styles = useStyles({ containerStyle, user });
  const { handleSubmitState, submitError, submitSuccessful, submitting } = useSubmitState();

  const verifiedText = user?.emailVerified
    ? 'Email Verified'
    : 'Email Not Verified   -';
  const verifiedIcon = user?.emailVerified
    ? 'check-circle-outline'
    : 'error-outline';

  return user && (isVisibleWhenVerified || !user.emailVerified) && (
    <View style={styles.container}>
      <View style={styles.statusContainer}>
        <Icon name={verifiedIcon} iconStyle={styles.icon} />
        <Text>{verifiedText}</Text>

        {!user.emailVerified && (
          <Button
            loading={submitting}
            onPress={handleSubmitState(() => user.sendEmailVerification())}
            titleStyle={styles.resendButton}
            title="Resend Email"
            type="clear"
          />
        )}
      </View>

      {submitSuccessful && (
        <Text style={styles.submitSuccessful}>
          Success - please check your email
        </Text>
      )}

      <ErrorText error={submitError} />
    </View>
  );
};

export type * from './EmailVerification.interfaces';
export default EmailVerification;
