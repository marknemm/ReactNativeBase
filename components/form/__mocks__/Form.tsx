import type { FormProps } from '@components/form/Form.interface';

const FormMock = jest.fn().mockImplementation((props: FormProps) => {
  const ActualComponent = jest.requireActual('@components/form/Form').default;
  return <ActualComponent {...props} />;
});

export default FormMock;
