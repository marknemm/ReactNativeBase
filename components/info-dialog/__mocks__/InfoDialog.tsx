import type { InfoDialogProps } from '@components/info-dialog/InfoDialog.interfaces';

const MockInfoDialog = jest.fn().mockImplementation((props: InfoDialogProps) => {
  const ActualComponent = jest.requireActual('@components/info-dialog/InfoDialog').default;
  return <ActualComponent {...props} />;
});

export default MockInfoDialog;
