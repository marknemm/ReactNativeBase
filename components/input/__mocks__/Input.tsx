import type { InputProps } from '@components/input/Input.interfaces';

const MockInput = jest.fn().mockImplementation((props: InputProps) => {
  const ActualComponent = jest.requireActual('@components/input/Input').default;
  return <ActualComponent {...props} />;
});

export default MockInput;
