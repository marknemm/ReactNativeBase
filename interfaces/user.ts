import { DBDocData } from './db';

/**
 * A raw {@link UserDoc} from a remote database.
 *
 * @extends DBDocData The {@link DBDocData} interface.
 */
export interface UserDoc extends DBDocData {

  /**
   * The user's mailing address.
   */
  address?: Address;

  /**
   * The user's display (full) name.
   */
  displayName?: string;

  /**
   * The user's email address.
   */
  email?: string;

  /**
   * The user's phone number.
   */
  phoneNumber?: string;

  /**
   * The user's photo URL.
   */
  photoURL?: string;

}

/**
 * A mailing {@link Address}.
 */
export interface Address {

  /**
   * The street address.
   */
  street: string;

  /**
   * The apartment or suite number.
   */
  apartmentSuite: string;

  /**
   * The city.
   */
  city: string;

  /**
   * The state.
   */
  state: string;

  /**
   * The ZIP code.
   */
  zip: string;

}
