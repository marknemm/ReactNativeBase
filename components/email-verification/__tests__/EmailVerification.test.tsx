import EmailVerification from '@components/email-verification/EmailVerification';
import AppProvider from '@test/contexts/app/AppProvider';
import { render, screen, userEvent } from '@testing-library/react-native';
import { getMockAuthUser } from '@util/__mocks__/user';
import User from '@util/user';

describe('<EmailVerification />', () => {
  let mockUser: User;

  beforeEach(() => {
    mockUser = getMockAuthUser();
    jest.spyOn(mockUser, 'emailVerified', 'get').mockReturnValue(true);
  });

  describe('verification status display', () => {
    it('displays nothing when verified and not configured to be visible when verified', () => {
      render(
        <EmailVerification user={mockUser} />,
        { wrapper: AppProvider }
      );

      const verifiedText = screen.queryByRole('text');
      expect(verifiedText).toBeFalsy();
    });

    it('displays verified text when configured to be visible when verified', () => {
      render(
        <EmailVerification
          isVisibleWhenVerified
          user={mockUser}
        />,
        { wrapper: AppProvider }
      );

      const verifiedText = screen.getByText(/Email Verified/i);
      expect(verifiedText).toBeTruthy();
    });

    it('displays unverified text when unverified', () => {
      jest.spyOn(mockUser, 'emailVerified', 'get').mockReturnValue(false);

      render(
        <EmailVerification user={mockUser} />,
        { wrapper: AppProvider }
      );

      const notVerifiedText = screen.getByText(/Email Not Verified/i);
      expect(notVerifiedText).toBeTruthy();
    });

    it('displays resend email button when unverified', () => {
      jest.spyOn(mockUser, 'emailVerified', 'get').mockReturnValue(false);

      render(
        <EmailVerification user={mockUser} />,
        { wrapper: AppProvider }
      );

      const resendEmailButton = screen.getByRole('button', { name: /Resend Email/i });
      expect(resendEmailButton).toBeTruthy();
    });
  });

  describe('resend email button', () => {
    it('sends email verification when resend email button is pressed', async () => {
      jest.spyOn(mockUser, 'emailVerified', 'get').mockReturnValue(false);

      render(
        <EmailVerification user={mockUser} />,
        { wrapper: AppProvider }
      );

      const resendEmailButton = screen.getByRole('button', { name: /Resend Email/i });
      await userEvent.press(resendEmailButton);

      expect(mockUser.sendEmailVerification).toHaveBeenCalledTimes(1);
    });

    it('displays success message when email verification is sent', async () => {
      jest.spyOn(mockUser, 'emailVerified', 'get').mockReturnValue(false);

      render(
        <EmailVerification user={mockUser} />,
        { wrapper: AppProvider }
      );

      const resendEmailButton = screen.getByRole('button', { name: /Resend Email/i });
      await userEvent.press(resendEmailButton);

      const successMessage = await screen.findByText(/Success - please check your email/i);
      expect(successMessage).toBeTruthy();
    });
  });

  describe('snapshots', () => {
    it('renders correctly when not visible', () => {
      render(
        <EmailVerification user={mockUser} />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly when verified and visible', () => {
      render(
        <EmailVerification
          isVisibleWhenVerified
          user={mockUser}
        />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });

    it('renders correctly when not verified', () => {
      jest.spyOn(mockUser, 'emailVerified', 'get').mockReturnValue(false);

      render(
        <EmailVerification user={mockUser} />,
        { wrapper: AppProvider }
      );

      expect(screen.toJSON()).toMatchSnapshot();
    });
  });
});
