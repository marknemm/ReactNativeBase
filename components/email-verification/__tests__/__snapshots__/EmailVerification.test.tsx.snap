// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailVerification /> snapshots renders correctly when not verified 1`] = `
<View
  style={{}}
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "overflow": "hidden",
        }
      }
      testID="RNE__ICON__CONTAINER"
    >
      <View
        accessibilityRole="button"
        android_ripple={
          {
            "borderless": false,
            "color": "rgba(0, 0, 0, 0.3)",
            "radius": -5,
          }
        }
        disabled={false}
        replaceTheme={[Function]}
        testID="RNE__ICON__CONTAINER_ACTION"
        updateTheme={[Function]}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
          testID="RNE__ICON"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#242424",
                  "fontSize": 24,
                },
                {
                  "backgroundColor": "transparent",
                  "color": "#ff3b30",
                  "marginRight": 3.142857142857143,
                },
                {
                  "fontFamily": "material",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            testID="RNE__ICON__Component"
          >
            
          </Text>
        </View>
      </View>
    </View>
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#242424",
          "fontFamily": undefined,
          "fontSize": 23.57142857142857,
          "fontStyle": "normal",
          "fontWeight": "normal",
          "marginBottom": 0,
          "textAlign": "left",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
        }
      }
    >
      Email Not Verified   -
    </Text>
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 3.142857142857143,
          },
          undefined,
          false,
        ]
      }
      testID="RNE_BUTTON_WRAPPER"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": false,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#007aff",
              "borderRadius": 3.142857142857143,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 12.571428571428571,
              "paddingHorizontal": 14.571428571428571,
            }
          }
        >
          <Text
            style={
              {
                "color": "#007aff",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "left",
              }
            }
          >
            Resend Email
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`<EmailVerification /> snapshots renders correctly when not visible 1`] = `null`;

exports[`<EmailVerification /> snapshots renders correctly when verified and visible 1`] = `
<View
  style={{}}
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "overflow": "hidden",
        }
      }
      testID="RNE__ICON__CONTAINER"
    >
      <View
        accessibilityRole="button"
        android_ripple={
          {
            "borderless": false,
            "color": "rgba(0, 0, 0, 0.3)",
            "radius": -5,
          }
        }
        disabled={false}
        replaceTheme={[Function]}
        testID="RNE__ICON__CONTAINER_ACTION"
        updateTheme={[Function]}
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            }
          }
          testID="RNE__ICON"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#242424",
                  "fontSize": 24,
                },
                {
                  "backgroundColor": "transparent",
                  "color": "#4cd964",
                  "marginRight": 3.142857142857143,
                },
                {
                  "fontFamily": "material",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
            testID="RNE__ICON__Component"
          >
            
          </Text>
        </View>
      </View>
    </View>
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#242424",
          "fontFamily": undefined,
          "fontSize": 23.57142857142857,
          "fontStyle": "normal",
          "fontWeight": "normal",
          "marginBottom": 0,
          "textAlign": "left",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
        }
      }
    >
      Email Verified
    </Text>
  </View>
</View>
`;
